components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -112,4 +117,51 @@ function AutomaticPointsTable:parseManualPoints(team, tournamentCount)
 	return manualPoints
 end
 
+function AutomaticPointsTable:queryPlacements(tournaments)
+	local queryParams = {
+		limit = 5000,
+		query = 'tournament, participant, placement, extradata'
+	}
+
+	local tournamentNames = Table.mapValues(tournaments,
+		function(tournament)
+			return tournament.name
+		end
+	)
+
+	local conditions = '[[tournament::' ..
+		table.concat(tournamentNames, ']] OR [[tournament::') ..
+		']]'
+
+	queryParams.conditions = conditions
+	local allQueryResult = mw.ext.LiquipediaDB.lpdb('placement', queryParams)
+
+	local tournamentsWithPlacements = Table.mapValues(tournaments,
+		function(tournament)
+			local filteredResult = Table.filter(allQueryResult,
+				function(result, tournamentName)
+					return result.tournament == tournamentName
+				end,
+				tournament.name
+			)
+
+			local tournamentPlacements = Table.mapValues(filteredResult,
+				function(result)
+					result.points = tonumber(result.extradata.prizepoints)
+					result.securedPoints = tonumber(result.extradata.securedpoints)
+					result.extradata = nil
+					return result
+				end
+			)
+
+			local tournamentWithPlacements = Table.deepCopy(tournament)
+			tournamentWithPlacements.placements = tournamentPlacements

```

Comment:
Here, I could alter the input and return it, but I've decided to use deepCopy to avoid doing that.